File: tests/js/aws-bedrock/test_manual.ts
/**
 * Conformance test: Manual memory instrumentation for AWS Bedrock.
 *
 * Exercises memory operations (create_memory_store, update_memory,
 * search_memory, delete_memory, delete_memory_store) with manual OTel spans.
 */

import { setupOtel, runMemoryOperations } from "./common";

async function main() {
  console.log("=== Manual: AWS Bedrock JS Memory Conformance Test ===");

  const { provider, meterProvider, loggerProvider } = setupOtel();

  await runMemoryOperations();

  console.log("Flushing telemetry...");
  await provider.forceFlush();
  await meterProvider.forceFlush();
  await loggerProvider.forceFlush();
  await provider.shutdown();
  await meterProvider.shutdown();
  await loggerProvider.shutdown();
  console.log("Done.");
}

main().catch((e) => { console.error(e); process.exit(1); });

File: tests/python/aws-bedrock/test_manual.py
"""Conformance test: AWS Bedrock manual memory instrumentation.

Exercises: memory operations (create_memory_store, update_memory,
search_memory, delete_memory, delete_memory_store) against a mock
Bedrock AgentCore server, with manual span instrumentation.
"""

from common import run, run_memory_operations


def instrument():
    """No-op: manual instrumentation only."""
    pass


if __name__ == "__main__":
    run(
        "Manual: AWS Bedrock Memory Conformance Test",
        instrument,
        [run_memory_operations],
    )
